### DISCUSSION TITLE
请问如何在编译时加入最新的核心？

### DISCUSSION THREAD

**Discussion by quanjindeng** [UnblockNeteaseMusic/luci-app-unblockneteasemusic#125] [Q&A] [answered]
我这边的网络不行，挂着梯子也会更新核心一直失败。
所以我想在编译的时候（GitHub云编译）就加入最新的核心进去，避免下载失败的问题。因为设定了几天就编译一次，这样也能用上最新的核心了。
请问要怎样操作呢？请求大佬解答，感谢！

**Comment by 1715173329** [comment, marked as answer]
luci-app-unblockneteasemusic/root/usr/share/unblockneteasemusic:
- core_local_ver
  写入[最新 commit sha](https://github.com/UnblockNeteaseMusic/server/tree/enhanced/precompiled)，如 `59ca6119b6d85543218c58a023266ae6930d0e1a`

- core/
  放入 [app.js](https://github.com/UnblockNeteaseMusic/server/raw/enhanced/precompiled/app.js) [ca.crt](https://github.com/UnblockNeteaseMusic/server/raw/enhanced/ca.crt) [server.crt](https://github.com/UnblockNeteaseMusic/server/raw/enhanced/server.crt) [server.key](https://github.com/UnblockNeteaseMusic/server/raw/enhanced/server.key) 四个文件

**Reply by quanjindeng** [reply]
**感谢，参照核心的升级脚本已经搞定了**

NAME=$"package/luci-app-unblockneteasemusic/root/usr/share/unblockneteasemusic" && mkdir -p $NAME/core

echo "$(uclient-fetch -qO- 'https://api.github.com/repos/UnblockNeteaseMusic/server/commits?sha=enhanced&path=precompiled' | jsonfilter -e '@[0].sha')">"$NAME/core_local_ver"

curl -L https://github.com/UnblockNeteaseMusic/server/raw/enhanced/precompiled/app.js -o $NAME/core/app.js
curl -L https://github.com/UnblockNeteaseMusic/server/raw/enhanced/precompiled/bridge.js -o $NAME/core/bridge.js
curl -L https://github.com/UnblockNeteaseMusic/server/raw/enhanced/ca.crt -o $NAME/core/ca.crt
curl -L https://github.com/UnblockNeteaseMusic/server/raw/enhanced/server.crt -o $NAME/core/server.crt
curl -L https://github.com/UnblockNeteaseMusic/server/raw/enhanced/server.key -o $NAME/core/server.key
